### PR TITLE
Extend IBAN validation

### PR DESCRIPTION
**Pull request by Corniel**
Add validation for countries with extended validation on IBAN's.

- [ ] **Albania**                   Weighted
- [ ] **Belgium**                   MOD-97-10 (variant)
- [x] **Bosnia and Herzegovina**    MOD-97-10
- [ ] **Croatia**                   MOD-11-10
- [ ] **Czech Republic**            Weighted
- [ ] **East Timor**                MOD-97-10
- [ ] **Estonia**                   Weighted
- [ ] **Finland**                   Luhn
- [ ] **France**                    MOD-97-10 (variant)
- [ ] **Hungary**                   Weighted
- [ ] **Iceland**                   Weighted
- [ ] **Italy**                     Conversion + Sum
- [ ] **Mauritania**                MOD-97-10 (variant)
- [x] **Mauritius**                  ends with currency
- [ ] **Monaco**                    MOD-97-10 (variant)
- [x] **Montenegro**                MOD-97-10
- [x] **North Macedonia**           MOD-97-10
- [ ] **Norway**                    Weighted
- [ ] **Poland**                    Weighted
- [x] **Portugal**                  MOD-97-10
- [ ] **San Marino**                Conversion + Sum
- [x] **Serbia**                    MOD-97-10
- [x] **Seychelles**               ends with currency
- [ ] **Slovakia**                  Weighted
- [x] **Slovenia**                  MOD-97-10
- [ ] **Spain**                     Weighted
- [x] **Tunisia**                   MOD-97-10 (variant)

### CODE REVIEW

**Comment by Corniel** [comment]
After some consideration, I decided not to implement (for now) the extended validation except Mauritius' and Seychelles'. Their validation (it should refer to an exciting currency code) is fundamental, and will not likely change. For the others: that validation was introduced prior to IBAN, and is, therefor, defacto obsolete. The Dutch authorities have officially dropped the Dutch 11-proof, and while implementing others (Hungary for example) It seemed that I could find actual IBAN's that did not longer satisfy the national checksum(s).

In other words: the risk of introducing false positives is way too high. Therefor, the impact of the PR has been reduced dramatically.
